Widgets_CDAT/textStyleEditorWidget/TextStyleEditor.py

```diff
@@ -0,0 +1,253 @@
+        self.HAGroup.addButton(C_button)
+        self.HAGroup.addButton(R_button)
+        self.HAGroup.buttonClicked.connect(self.updateButton)
+
+        HABox = QtGui.QGroupBox()
+        HABox.setLayout(HALayout)
+        HABox.setTitle("Horizontal Align")
+
+        # First row
+        align_angle_row = QtGui.QHBoxLayout()
+        align_angle_row.addWidget(VABox)
+        align_angle_row.addWidget(HABox)
+        align_angle_row.insertStretch(2,1)
+
+        # Preview setup
+        self.preview = text_style_preview.TextStylePreviewWidget()
+
+        # Create labels
+        angle_label = QtGui.QLabel()
+        angle_label.setText("Angle:")
+        font_label = QtGui.QLabel()
+        font_label.setText("Font:")
+        size_label = QtGui.QLabel()
+        size_label.setText("Size:")
+
+        # angle dial setup
+        self.angle_slider = QtGui.QDial()
+        self.angle_slider.setRange(90, 450)
+        self.angle_slider.setWrapping(True)
+        self.angle_slider.setNotchesVisible(True)
+        self.angle_slider.valueChanged.connect(self.updateAngle)
+
+        align_angle_row.addWidget(angle_label)
+        align_angle_row.addWidget(self.angle_slider)
+
+        # Font combobox
+        self.font_box = QtGui.QComboBox()
+        for item in vcs.listelements("font"):
+            self.font_box.addItem(item)
+        self.font_box.currentIndexChanged[str].connect(self.updateFont)
+
+        # size spin box
+        self.size_box = QtGui.QSpinBox()
+        self.size_box.setRange(1, 128)
+        self.size_box.valueChanged.connect(self.updateSize)
+
+
+        font_size_row = QtGui.QHBoxLayout()
+        font_size_row.addWidget(font_label)
+        font_size_row.addWidget(self.font_box)
+        font_size_row.addWidget(size_label)
+        font_size_row.addWidget(self.size_box)
+        font_size_row.insertStretch(2, 3)
+
+        # Save and Cancel Buttons
+
+        cancel_button = QtGui.QPushButton()
+        cancel_button.setText("Cancel")
+        cancel_button.clicked.connect(lambda: self.close())
+
+        saveas_button = QtGui.QPushButton()
+        saveas_button.setText("Save As")
+        saveas_button.clicked.connect(self.saveAs)
+
+        save_button = QtGui.QPushButton()
+        save_button.setText("Save")
+        save_button.clicked.connect(self.save)
+
+        save_cancel_row = QtGui.QHBoxLayout()
+        save_cancel_row.addWidget(cancel_button)
+        save_cancel_row.addWidget(saveas_button)
+        save_cancel_row.addWidget(save_button)
+        save_cancel_row.insertStretch(1, 1)
+
+        # Set up wrap
+        wrap.addWidget(self.preview)
+        wrap.addLayout(align_angle_row)
+        wrap.addLayout(font_size_row)
+        wrap.addLayout(save_cancel_row)
+        self.setLayout(wrap)
+
+
+    def setTextObject(self, textObject):
+        self.textObject = textObject
+        self.preview.setTextObject(self.textObject)
+        self.setWindowTitle("Edit Style \"" + self.textObject.name + "\"")
```

Comment:
You can use single quotes to avoid having to escape the double quote:

```python
self.setWindowTitle('Edit Style "' + self.textObject.name + '"')
```

though the preferred style of doing a task like this would be:

```python
self.setWindowTitle('Edit Style "%s"' % self.textObject.name)
```

or (this is preferred when you're doing a bunch of insertions) 

```python
self.setWindowTitle('Edit Style "{name}"'.format(name=self.textObject.name))
```